WIP/calculator.rb

```diff
@@ -0,0 +1,104 @@
+class CheckoutSummary
+
+  attr_reader :discounts
+
+  #the default value for accumulate is true unless specified or set to be false
+  def initialize(**args)
+    @accumulate = args[:accumulate] || args[:accumulate] == nil
+    @discounts = {}
+    @gross_cost = args[:gross_cost]
+  end
+
+  def add_discount(**args)
+
+    name = args[:name]
+    amount = args[:amount]
+    percentage = args[:percentage]
+
+
+    if amount.is_a?(String) || percentage.is_a?(String) || amount.to_i < 0  
+      raise StandardError.new "Not a valid deal"
+    elsif amount == nil && percentage == nil
+      raise StandardError.new "Not a valid deal"
+    elsif percentage.to_i > 100 || percentage.to_i < 0
+      raise StandardError.new "Not a valid deal"
+    end
+
+    @discounts[name] = {
+      name: name,
+      amount: amount,
+      percentage: percentage
+    }
+
+    true
+  end
+
+  
+
+  def accumulate?
+    @accumulate
+  end
+
+  def remove_discount!(discount_name)
+    @discounts.delete(discount_name)
+  end
+
+  
+  def calculate_discount_accumulate_true
```

Comment:
`calculate_discount_accumulate_true` and `calculate_discount_accumulate_false` could be private methods as they are not available to be called directly